ARCHITECTURE.md

```diff
@@ -34,6 +34,12 @@ To set up a development machine, do the following:
 
     This is important so that each service in the compose file can have a hostname.
 
+1. Get the Gateway IP of the new bridge network and paste it in the settings file:
+
+    `docker network inspect bc2_net_overlay --format "{{range .IPAM.Config}}{{.Gateway}}{{end}}"`
+
+    This is important so We can use the `debug` tag in the templates.
```

Comment:
I'm not familiar with this. Is there a reference you could share or that we could link to here?

Reply:
I'm going to try to summarize what I learn from a few articles about the `debug` tag, the django `INTERNAL_IPS` setting, and the docker documentation about bridge networks.

In this PR, We include `HTMX` to the HTML templates, and We'd like to use the **minified** version for production and also supply  the non-minified versions to our developers(if they're available), so you suggested We can do something like: 

``` python
{% if debug %}
    src=non-minified-version.js
{% else %}
     src=this-version.min.js
{% endif %}
```

The previous `if statement` seems pretty easy to implement since django has a tag called `debug`, but after checking the docs I found out We also need to set the `INTERNAL_IPS` to let Django know that its ok to disclose sensitive information( like the `debug` variable) to the template's context. 

I checked the `setting` module and realized the `INTERNAL_IPS` variable was defined in the security.py file. We added just `127.0.0.1` to the list, but the `if statement` was still not working. Something was wrong with the list. I noticed some of the projects that suggest values for `INTERNAL_IPS`(like [django-tailwind](https://github.com/timonweb/django-tailwind)) didn't use Docker to set up the development environment, so I started looking at the Docker docs about networking.

I found that: 

> By default, a container gets an IP address for every Docker network it attaches to. A container receives an IP address out of the IP pool of the network it attaches to and each network also has a default subnet mask and gateway.

so the request originating from the `localhost` is passed through the **Gateway** of the network, due to which the Django application resolves the originating IP address of the request as the **Gateway IP** of the Docker network, this means that even if we have the set ["127.0.0.1"] as `INTERNAL_IPS` during development in the local machine, the django app will not add the debug variable to the context because the **Gateway IP** is not included in the list.

Reply:
While I was looking for the links I reviewed to come up with this, I found that [django-debug-toolbar](https://github.com/jazzband/django-debug-toolbar) and [cookiecutter-django](https://github.com/cookiecutter/cookiecutter-django) implements a simpler and easier way to get the **Gateway IP** using the socket module.

They have the following code in the setting file:

```
import socket

hostname, _, ips = socket.gethostbyname_ex(socket.gethostname())
INTERNAL_IPS += [".".join(ip.split(".")[:-1] + ["1"]) for ip in ips]
```

in the previous code, we have a few operations: 

1. `socket.gethostbyname_ex(socket.gethostname())` returns the IP address of the host machine in the ips variable.

2. `[ip[:-1] + '1' for ip in ips]` removes the last digit of the host IP address and replaces it with 1. This is done because the host container receives its IP address out of the IP pool of the network and the pool usually uses the X.X.X.1 address as a default gateway (We could change the default gateway but that's not our case, We don't use the advanced options in the `docker network create` command). let's say the IP address of the host is 172.17.0.3 by default, the gateway of the Docker network is set 172.17.0.1, hence the last value of the IP address can be replaced with 1 to get the default gateway.

We can use this code and remove the extra step for developers..

Reply:
let me know what you think.

Reply:
Well, that explains why I haven't been able to get the debug toolbar to work for a while. Let's do their approach and shall we install the debug toolbar too? We should probably get it working in CL again, if you're down in this area of thinking and it's easy to do.